### PR TITLE
Fix: sort icons to group variants

### PR DESCRIPTION
**Pull request by tomasciccola**
given that this bug wasn't caught by tests, I use this PR to add a couple more tests of icons

### CODE REVIEW

File: src/config-import.js [hunk 1]
```diff
@@ -65,10 +57,12 @@ export async function readConfig(configPath) {
       /** @type {IconData | undefined} */
       let icon
 
-      iconsZip = await yauzl.open(configPath)
-      const entries = await iconsZip.readEntries(MAX_ENTRIES)
-      for (const entry of entries) {
-        if (!entry.filename.match(/^icons\/([^/]+)$/)) continue
+      // we sort the icons by filename so we can group variants together
+      const iconEntries = entries
+        .filter((entry) => entry.filename.match(/^icons\/([^/]+)$/))
```
Comment (gmaclennan) [inline src/config-import.js:62]:
```suggestion
        .filter((entry) => entry.filename.startsWith('icons/')
```

Fine as it is, just not sure the regex is necessary

File: src/config-import.js [hunk 2]
```diff
@@ -65,10 +57,12 @@ export async function readConfig(configPath) {
...
+        .sort((icon, nextIcon) => (icon.filename > nextIcon.filename ? 1 : -1))
```
Comment (gmaclennan) [inline src/config-import.js:63]:
```suggestion
        .sort((icon, nextIcon) => icon.filename.localeCompare(nextIcon.filename))
```
Not a necessary change, but just sharing this string method if you don't know about it - can be handy for this kind of logic. However note that your current implementation is missing the case when the name is equal, which should return `0`, but this is ok given that it should not be possible for this array to include two entries with the same filename - just pointing out that you would normally need this to ensure a stable sort order I think.

File: tests/config-import.js [hunk 3]
```diff
@@ -131,12 +131,12 @@ test('config import - icons', async (t) => {
...
-      icon.variants.length,
-      9,
-      '9 variants of icons (dot product of density and size)'
-    )
+    // t.is(icon.name, 'plant', 'icon name is `plant`')
+    if (icon.name === 'plant') {
+      t.is(icon.variants.length, 3, '3 variants of plant icons')
+    } else if (icon.name === 'tree') {
+      t.is(icon.variants.length, 9, '9 - all - variants of tree icons')
+    }
```
Comment (gmaclennan) [inline tests/config-import.js:139]: This test would pass if the fixture happened to not include icons with those names. The test should check that these assertions have been called - normally done with a `t.plan()`.

**Review by gmaclennan — CHANGES_REQUESTED** [review]
Just a couple of nit-picks on the code, which don't actually need any change, since they do work fine as-is.

The test though needs a check to make sure that your new assertions are actually being called.

File: src/config-import.js (hunk 1 above)
Comment (tomasciccola) [reply src/config-import.js:62]: The difference is that with the regex you only match files in the icon folder, where with the other option you will also match with 'icons/' (so the listing of the directory itself) which we don't want. Instead, we could do:
```js
.filter(
          (entry) =>
            entry.filename.startsWith('icons/') && !entry.filename.endsWith('/')
        )

```
but I dunno...

File: src/config-import.js (hunk 2 above)
Comment (tomasciccola) [reply src/config-import.js:63]: uhh I like local compare! prefer it much more that comparing strings as if they were 'numbers'

File: src/config-import.js (hunk 1 above)
Comment (gmaclennan) [reply src/config-import.js:62]: Yeah, just keep the regex

**Pull request merged**